model/LtiResultAliasStorage.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017  (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\ltiDeliveryProvider\model;
+
+use oat\taoDelivery\model\execution\DeliveryExecution;
+use oat\oatbox\service\ConfigurableService;
+
+/**
+ * Class LtiResultAliasStorage
+ * @package oat\ltiDeliveryProvider\model
+ * @author Aleh Hutnikau, <hutnikau@1pt.com>
+ */
+class LtiResultAliasStorage extends ConfigurableService
+{
+    const OPTION_PERSISTENCE = 'persistence';
+
+    const SERVICE_ID = 'ltiDeliveryProvider/LtiResultIdStorage';
+
+    const TABLE_NAME = 'lti_result_identifiers';
+    const DELIVERY_EXECUTION_ID = 'delivery_execution_id';
+    const RESULT_ID = 'result_id';
+
+    /** @var \common_persistence_SqlPersistence */
+    protected $persistence;
+
+    /**
+     * @return string
+     */
+    public function getTableName()
+    {
+        return self::TABLE_NAME;
+    }
+
+    /**
+     * Add record to the storage
+     * @param DeliveryExecution $deliveryExecution
+     * @param string $resultId
+     * @return boolean
+     */
+    public function log(DeliveryExecution $deliveryExecution, $resultId)
```

Comment:
This could be a generic functionality:
`public function addResultAlias($deliveryExecutionId, $alias);`